HealthMonitoring.AcceptanceTests/Scenarios/Selenium/Home_page.Steps.cs

```diff
@@ -114,27 +123,27 @@ public void Then_status_filter_should_be_appended_to_url()
         {
             var selectedStatus = GetSelectedStatusElements().First();
             var expectedUrl = $"{_homeUrl}&filter-status={selectedStatus.Text}";
-            var actualUrl = _driver.WaitUntilPageIsChanged(_homeUrl);
+            var actualUrl = _driver.WaitUntilPageIsChanged(expectedUrl);
 
             CustomAssertions.EqualNotStrict(actualUrl, expectedUrl);
         }
 
-        public void When_user_clicks_on_endpoint_tags()
+        public void When_user_clicks_on_first_tag()
         {
-            _selectedTags = new List<string>();
-
-            var allTags = GetAllTags();
-            var firstTag = allTags.First();
-
-            firstTag.Click();
+            var firstTag = GetAllTags().First();
             _selectedTags.Add(firstTag.Text);
+            firstTag.Click();
 
-            allTags = GetAllTags();
-            var secondTag = allTags.First(m => m.Text != _selectedTags.First());
+            Thread.Sleep(3000);
```

Comment:
I could not find better way to prevent races between test client and browser in this test.